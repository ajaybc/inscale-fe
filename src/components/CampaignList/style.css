.CampaignList {
}

.CampaignList__table {
  width: 100%;
  border: 1px solid #dee2e6;
  border-collapse: collapse;
}

.CampaignList__table th {
  font-weight: bold;
}

.CampaignList__table th, .CampaignList__table td {
  border: 1px solid #dee2e6;
  padding: 0.75rem;
  text-align: left;
}

.CampaignList__table > thead {
}

.CampaignList__pagination {
  padding-top: 10px;
}

.CampaignList__pagination li{
  display: inline-block;
}

.CampaignList__pagination li > a{
  display: block;
  padding: 10px;
  background-color: #fff;
  border: 1px solid #dee2e6;
}

.CampaignList__pagination li.previous > a{
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.CampaignList__pagination li.next > a{
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.CampaignList__pagination li.break{
  padding-left: 5px;
  padding-right: 5px;
  color: #CCC;
}