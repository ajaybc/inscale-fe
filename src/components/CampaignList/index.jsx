import React, {Component} from 'react';
import ReactPaginate from 'react-paginate';

import 'react-datepicker/dist/react-datepicker.css';
import './style.css';
import Status from '../Status';
import Price from '../Price';

export default class CampaignList extends Component {
  render () {
    const { campaigns } = this.props;
    return <div className="CampaignList">
      <table className="CampaignList__table">
        <thead>
          <tr>
            <th>Name</th>
            <th>Start Date</th>
            <th>End Date</th>
            <th>Active</th>
            <th>Budget</th>
          </tr>
        </thead>
        <tbody>
          {
            campaigns.map((campaign) => {
              return <tr key={campaign.id} >
                <td>{campaign.name}</td>
                <td>{campaign.startDate}</td>
                <td>{campaign.endDate}</td>
                <td><Status startDate={campaign.startDate} endDate={campaign.endDate}/></td>
                <td><Price>{campaign.Budget}</Price></td>
              </tr>
            })
          }
        </tbody>
      </table>
      <div className="CampaignList__pagination">
        <ReactPaginate 
          pageCount={100}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          forcePage={50}
          containerClassName={"pagination"}
          subContainerClassName={"pages pagination"}
          activeClassName={"active"}
          // onPageChange={this.handlePageClick}
          />
      </div>
    </div>;
  }
}