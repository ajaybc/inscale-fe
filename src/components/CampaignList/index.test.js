import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';

import CampaignList from './index.jsx';

const campaigns = [
  {
    "id": 1,
    "name": "Divavu",
    "startDate": moment('1989-08-11'),
    "endDate": moment('2018-08-11'),
    "Budget": 88377
  },
  {
    "id": 2,
    "name": "Jaxspan",
    "startDate": moment('1989-08-12'),
    "endDate": moment('2018-08-12'),
    "Budget": 608715
  },
  {
    "id": 3,
    "name": "Miboo",
    "startDate": moment('1989-08-13'),
    "endDate": moment('2018-08-13'),
    "Budget": 2395073232
  }
];

it('CampaignList renders correctly when campaigns are provided', () => {
  const tree = renderer
    .create(<CampaignList campaigns={campaigns} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('CampaignList renders correctly when campaigns are empty', () => {
  const tree = renderer
    .create(<CampaignList campaigns={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});