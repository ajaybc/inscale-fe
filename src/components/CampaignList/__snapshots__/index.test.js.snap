// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignList renders correctly when campaigns are empty 1`] = `
<div
  className="CampaignList"
>
  <table
    className="CampaignList__table"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th
          width="20%"
        >
          Start Date
        </th>
        <th
          width="20%"
        >
          End Date
        </th>
        <th
          width="15%"
        >
          Active
        </th>
        <th
          width="15%"
        >
          Budget
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="5"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          No campaigns found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CampaignList renders correctly when campaigns are provided 1`] = `
<div
  className="CampaignList"
>
  <table
    className="CampaignList__table"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th
          width="20%"
        >
          Start Date
        </th>
        <th
          width="20%"
        >
          End Date
        </th>
        <th
          width="15%"
        >
          Active
        </th>
        <th
          width="15%"
        >
          Budget
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Divavu
        </td>
        <td>
          11/08/1989
        </td>
        <td>
          11/08/2018
        </td>
        <td>
          <div
            className="Status__status Status__status--active"
          />
           
          active
        </td>
        <td>
          88.377K USD
        </td>
      </tr>
      <tr>
        <td>
          Jaxspan
        </td>
        <td>
          12/08/1989
        </td>
        <td>
          12/08/2018
        </td>
        <td>
          <div
            className="Status__status Status__status--active"
          />
           
          active
        </td>
        <td>
          608.715K USD
        </td>
      </tr>
      <tr>
        <td>
          Miboo
        </td>
        <td>
          13/08/1989
        </td>
        <td>
          13/08/2018
        </td>
        <td>
          <div
            className="Status__status Status__status--active"
          />
           
          active
        </td>
        <td>
          2395.073232M USD
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
